go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -171,33 +162,36 @@ func (bnc *BaseNetworkController) ensurePod4SecondaryNetwork(pod *kapi.Pod, addP
 
 // removePod tried to tear down a pod. It returns nil on success and error on failure;
 // failure indicates the pod tear down should be retried later.
-func (bnc *BaseNetworkController) removePod4SecondaryNetwork(pod *kapi.Pod, portInfo *lpInfo) error {
-	var nadName, logicalPortName string
+func (bnc *BaseNetworkController) removePod4SecondaryNetwork(pod *kapi.Pod, portInfoMap map[string]*lpInfo) error {
+	var nadName string
+	var portInfo *lpInfo
 
-	if !util.PodWantsNetwork(pod) {
-		return nil
-	}
 	podDesc := pod.Namespace + "/" + pod.Name
 	klog.Infof("Deleting pod: %s for network %s", podDesc, bnc.GetNetworkName())
 
-	if !util.PodScheduled(pod) {
-		return nil
-	}
+	if portInfoMap != nil {
```

Comment:
In which cases would we expect this to be nil? Can you add a comment for that?

Reply:
there are two cases, first case is that addLogicalPortToNetwork failed in between and portInfo is not added to the logicalPortCache, second case is when ovnkube-master restarts, and logicalPortCache is not initailized yet.

When portInfoMap is nil, we should not check isNetworkOnPod(), instead, we could unmarshall pod annotations to find any nad belongs to this network and release resources then.

Reply:
we'd still need to know the nadName before calling deletePodLogicalPort()